common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -541,6 +544,51 @@ public Object run() {
                 DIRECT_BUFFER_CONSTRUCTOR != null ? "available" : "unavailable");
     }
 
+    private static Method getVirtualThreadCheckMethod() {
+        if (JAVA_VERSION < 19) {
+            return null;
+        }
+        try {
+            Method isVirtualMethod = Thread.class.getMethod("isVirtual");
+            // Call once to make sure the invocation works.
+            boolean isVirtual = (Boolean) isVirtualMethod.invoke(Thread.currentThread());
+            return isVirtualMethod;
+        } catch (Throwable e) {
+            if (logger.isTraceEnabled()) {
+                logger.debug("virtual thread check method is not available: ", e);
+            } else {
+                logger.debug("virtual thread check method is not available: ", e.getMessage());
+            }
+            return null;
+        }
+    }
+
+    /**
+     * @param thread the thread to be checked.
+     * @return
+     * {@code 0}: MUST be a virtual thread.
+     * <br>
+     * {@code 1}: MUST NOT be a virtual thread.
+     * <br>
+     * {@code -1}: Not able to check the thread type.
+     */
+    static int checkVirtualThread(Thread thread) {
+        if (JAVA_VERSION < 19 || thread instanceof FastThreadLocalThread) {
+            return 1;
+        }
+        if (VIRTUAL_THREAD_CHECK_METHOD == null) {
+            return -1;
+        }
+        try {
+            return (Boolean) VIRTUAL_THREAD_CHECK_METHOD.invoke(thread) ? 0 : 1;
```

Comment:
Please use the `async` profiler plugin for JMH since this should be pretty hot and IIRC, without `MethodHandle::invokeExact` this is going to impact all worker pool use cases - which is not great

Reply:
> this is going to impact all worker pool use cases - which is not great

The event-loop thread(`FastThreadLocalThread`) will never reach`invoke()`, it's already guarded by 
https://github.com/netty/netty/blob/1462bb16257842ee971f4e5eec33f787f5bd36fc/common/src/main/java/io/netty/util/internal/PlatformDependent0.java#L575-L578